File: Linguagem Java/Questão 2.java
import java.util.Scanner;

public class VerificadorAprovacao {

    public static void main(String[] args) {
   
        Scanner scanner = new Scanner(System.in);

    
        System.out.print("Digite a primeira nota (0 a 100): ");
        float nota1 = scanner.nextFloat();

        System.out.print("Digite a segunda nota (0 a 100): ");
        float nota2 = scanner.nextFloat();

    
        float media = (nota1 + nota2) / 2;

      
        if (media >= 70) {
            System.out.println("Aprovado! Média: " + media);
        } else if (media >= 40 && media < 70) {
  
            System.out.println("Em exame. Média: " + media);

   
            System.out.print("Digite a nota do exame (0 a 100): ");
            float notaExame = scanner.nextFloat();

   
            if ((media + notaExame) >= 100) {
                System.out.println("Aprovado! Média final: " + (media + notaExame));
            } else {
                System.out.println("Reprovado. Média final: " + (media + notaExame));
            }
        } else {
   
            System.out.println("Reprovado. Média: " + media);
        }

        
        scanner.close();
    }
}

File: Linguagem Python/Questão 2.py
nota1 = int(input("Informe sua nota de 0 a 100: "))
nota2 = int(input("Informe sua segunda nota de 0 a 100:"))
media = (nota1 + nota2)/2
if (media >=70):
    print("Parabéns!!! Aprovado")
elif (media >= 40):
    print("Quase!!! Terá que fazer exame final")
    notaFinal = int(input("Informe a nota do exame final: "))
    if (notaFinal >= 70):
        print(f'Parabéns!!! Aprovado com {notaFinal}')
else:
    print("Uma pena!! Está reprovado")
